### PR TITLE
[WIP] internal whitelist

### PR DESCRIPTION
**Pull request by q3k**
Hic sunt dracones. This is not a real PR.

### CODE REVIEW

**Comment by q3k** [comment]
Posting odd integration test failure, as per @fisx 's request.
```
$ make integration-whitelist
Brig API Integration
  user
    account
      post /register - 403 whitelist:                   FAIL
        Exception: Assertions failed:
         1: 403 =/= 400
        
        Response was:
        
        Response {responseStatus = Status {statusCode = 400, statusMessage = "Bad Request"}, responseVersion = HTTP/1.1, responseHeaders = [("Content-Type","application/json")], responseBody = Just "{\"code\":400,\"message\":\"Error in $: not enough input\",\"label\":\"bad-request\"}", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}
        CallStack (from HasCallStack):
          error, called at src/Bilge/Assert.hs:69:30 in bilge-0.22.0-jGRJFHqHkhHCMotYIfBUv:Bilge.Assert
          <!!, called at src/Bilge/Assert.hs:86:19 in bilge-0.22.0-jGRJFHqHkhHCMotYIfBUv:Bilge.Assert
          !!!, called at test/integration/API/User/Account.hs:321:9 in main:API.User.Account
```

File: services/brig/test/integration/API/User/Account.hs [hunk 1]
```diff
@@ -312,6 +313,21 @@ testCreateUserBlacklist brig aws =
...
+testCreateUserWhitelist :: Brig -> Maybe Opt.Opts -> Http ()
+testCreateUserWhitelist _ Nothing = error "no options"
+testCreateUserWhitelist brig (Just opts) = do
+    e <- randomEmail
+    withSettingsOverrides whitelistEnabled $ do
+        post (brig . path "/register" . contentJson . body (p e)) !!! const 403 === statusCode
+    return ()
+  where
+    whitelistEnabled :: Opt.Opts
+    whitelistEnabled = opts & Opt.optionSettings . Opt.internalWhitelist .~ (Just $ Opt.InternalWhitelist [])
```
Comment (fisx) [inline services/brig/test/integration/API/User/Account.hs]:
```suggestion
    whitelistEnabled = opts & Opt.optionSettings . Opt.internalWhitelist .~ (Just $ Opt.InternalWhitelist ["Bob"])
```

(not sure that's much of a difference, but that's what i would have done.)

**Comment by mheinzel** [comment]
For future readers: The source of the odd integration test failure got fixed in https://github.com/wireapp/wire-server/pull/969.

**Pull request closed without merging**